pkg/sharding/ranges.go

```diff
@@ -119,3 +131,30 @@ func (l *LogRanges) String() string {
 	ranges = append(ranges, fmt.Sprintf("active=%d", l.active))
 	return strings.Join(ranges, ",")
 }
+
+// PublicKey returns the associated public key for the given Tree ID
+// and returns the active public key by default
+func (l *LogRanges) PublicKey(activePublicKey, treeID string) (string, error) {
```

Comment:
Is the `activePublicKey` going to be different each time this is called? How will this normally be specified?

Reply:
it should be the same every time, we get it from the signer passed in via the `--rekor_server.signer` flag

Reply:
Would it make sense to make it global then instead of an argument? Just thinking out loud / nitpicking, it is also fine as-is.

Reply:
Yah that would be better! I tried to do that initially but it required a bigger refactor since right now the the log ranges are passed in to `NewAPI` and that's where we get the public key. We'd need to move the public key logic out for this to work & I thought about doing it in another PR so this one wouldn't get too big.

Reply:
That's totally reasonable. Thanks.